### PR TITLE
feat: support non-indented cases in Scala3 match expression

### PR DESCRIPTION
**Pull request by susliko**
Addresses #264

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1157,7 +1162,7 @@ module.exports = grammar({
         optional($.inline_modifier),
         field("value", $.expression),
         "match",
-        field("body", choice($.case_block, $.indented_cases)),
+        field("body", choice($.case_block, $.indented_cases, $.non_indented_cases)),
```
Comment (susliko) [inline grammar.js:1165]: Should `non_indented_cases` be aliased to `indented_cases` here to preserve compatibility for any queries that may not expect nodes other than `case_block`/`indented_cases`, wdyt?

**Comment by susliko** [comment]
Looks like CI ran against an old version, without changes in grammar.js. Strange 🤔

File: test/corpus/expressions.txt [hunk 2]
```diff
@@ -850,8 +866,8 @@ class C {
     a
       .b
       // comment1
-      /* 
-         comment2 
+      /*
```
Comment (susliko) [inline test/corpus/expressions.txt:869]: Changes here and below are caused by changes in behavior of `tree-sitter test -u`, which now removes trailing spaces

File: grammar.js (hunk 1 above)
Comment (eed3si9n) [reply grammar.js:1165]: +1 to this.

**Comment by eed3si9n** [comment]
> Looks like CI ran against an old version, without changes in grammar.js. Strange 🤔

Previously, we were running generate explicitly only on Linux, but at some point CI started testing on macOS as well? I fixed the CI in https://github.com/tree-sitter/tree-sitter-scala/pull/437

**Review by eed3si9n — APPROVED** [review]
LGTM

**Comment by eed3si9n** [comment]
```
Checking syntax complexity: npm exec -c 'tree-sitter generate --report-states-for-rule compilation_unit' 2>&1 >/dev/null
Report written to report-complexity.txt
tuple_type 	1857
Error: complexity of the most complex definition tuple_type: 	1857, higher than the allowed ceiling 1400
```

I guess PREC.right adds to the complexity a bit? I'm ok with bumping this up.

**Comment by susliko** [comment]
While working on this PR, I realized that the conflict between `$.match_expression` and `$._block` is slowing down the generation process by about 9 times. I haven't found a good solution yet, so I'm going to close the PR for now.

**Pull request closed without merging**